gridfs_bucket.go

```diff
@@ -105,26 +103,12 @@ func (b *gridfsBucket) Join(elems ...string) string { return consistentJoin(elem
 // supports a single read and a single write deadline for a bucket instance.
 // This function sets the read and write deadlines to allow it to respect the
 // context timeouts passed in by the caller.
-func (b *gridfsBucket) bucket(ctx context.Context) (*gridfs.Bucket, error) {
+func (b *gridfsBucket) bucket(ctx context.Context) (*mongo.GridFSBucket, error) {
```

Comment:
Could remove the `ctx` parameter since it's not used anymore.

Reply:
If the context errors, the later ops in the functios that pass the ctx will error anyways. In general, manually and explicitly checking `ctx.Err()` is only useful in a select few situations.